ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/LoadingPaywallView.kt

```diff
@@ -0,0 +1,148 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.material3.ColorScheme
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.ViewModel
+import com.revenuecat.purchases.Offering
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.PackageType
+import com.revenuecat.purchases.models.Period
+import com.revenuecat.purchases.models.Price
+import com.revenuecat.purchases.paywalls.PaywallData
+import com.revenuecat.purchases.ui.revenuecatui.data.PaywallViewModel
+import com.revenuecat.purchases.ui.revenuecatui.data.PaywallViewState
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.PaywallTemplate
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.TemplateConfiguration
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.VariableDataProvider
+import com.revenuecat.purchases.ui.revenuecatui.data.testdata.MockApplicationContext
+import com.revenuecat.purchases.ui.revenuecatui.data.testdata.TestStoreProduct
+import com.revenuecat.purchases.ui.revenuecatui.extensions.createDefault
+import com.revenuecat.purchases.ui.revenuecatui.helpers.toAndroidContext
+import com.revenuecat.purchases.ui.revenuecatui.helpers.toPaywallViewState
+import com.revenuecat.purchases.ui.revenuecatui.templates.Template2
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import java.net.URL
+
+@Composable
+internal fun LoadingPaywallView(mode: PaywallViewMode) {
+    val paywallData: PaywallData = PaywallData.createDefault(
+        LoadingPaywallConstants.packages,
+        MaterialTheme.colorScheme,
+    )
+
+    val offering = Offering(
+        identifier = LoadingPaywallConstants.offeringIdentifier,
+        serverDescription = "Loading paywall",
+        metadata = emptyMap(),
+        availablePackages = LoadingPaywallConstants.packages,
+        paywall = paywallData,
+    )
+
+    val viewModel = LoadingViewModel(mode, offering, paywallData)
+
+    val state = offering.toPaywallViewState(
+        VariableDataProvider(LocalContext.current.applicationContext.toAndroidContext()),
+        mode,
+        paywallData,
+        LoadingPaywallConstants.template,
+    )
+
+    when (state) {
+        is PaywallViewState.Error,
+        is PaywallViewState.Loading,
+        -> Box {}
+        is PaywallViewState.Loaded -> Template2(
+            state = state,
+            viewModel = viewModel,
+        )
+    }
+}
+
+private object LoadingPaywallConstants {
+    const val offeringIdentifier = "loading_offering"
+
+    val template = PaywallTemplate.TEMPLATE_2
+
+    val packages = listOf(
+        Package(
+            identifier = "weekly",
+            packageType = PackageType.WEEKLY,
+            offering = this.offeringIdentifier,
+            product = TestStoreProduct(
```

Comment:
Hmm not a fan of using `TestStoreProduct` in production... But I guess we would have to do something similar in any case for the loading screen following this route...

I'm wondering whether we would like to have a completely separate loading screen which doesn't need actual data... But I guess that would mean duplicating a lot of the UI layout, so probably not be worth it. I think we can go with this for now.

Reply:
Yeah it's slightly weird, but are you worried about something in particulaar?

Reply:
Well, mostly worried about using these test models in production... I think it's a bit of a smell TBH... But yeah, as long as we make sure it's limited to the `LoadingPaywallView`, it should be ok for now I think. We can reconsider about adding a separate layout for the loading view later if needed.